clientv3/client_test.go

```diff
@@ -49,7 +49,13 @@ func TestDialCancel(t *testing.T) {
 	c.SetEndpoints("http://254.0.0.1:12345")
 
 	// issue Get to force redial attempts
-	go c.Get(context.TODO(), "abc")
+	getc := make(chan struct{})
+	go func() {
+		defer close(getc)
+		// Get may hang forever on grpc's Stream.Header() if its
+		// context is never canceled.
+		c.Get(c.Ctx(), "abc")
```

Comment:
I was getting timeouts of up to 20 seconds. I didn't bother checking for longer. It's hanging on this select: https://github.com/coreos/etcd/blob/master/cmd/vendor/google.golang.org/grpc/transport/transport.go#L238. I don't think it's related to the balancer.